### PR TITLE
finalizeSaltTables

### PR DESCRIPTION
**Pull request by goncalvesmachadoc**

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

File: tests/parser/SaltTableTests.cpp [hunk 1]
```diff
@@ -53,26 +56,40 @@ BOOST_AUTO_TEST_CASE( Brine ) {
...
+        " 1000 0.0/\n"
         " 0 1 2 3 4 \n"
         " 10 11 12 13 14/\n"
-        " \n"
+        "\n"  
+        "RWGSALT\n"
+        " 300 0.0 0.00013/ \n"
+        " 600 0.5 0.000132/ \n"
+        "/ \n"  
+        "PERMFACT\n"
```
Comment (joakim-hove) [inline tests/parser/SaltTableTests.cpp:67]: The `PERMFACT` keyword as specified in json does not have a `size:` setting - that implies that it should have default size (i.e. number of records) behavior, which again means that the whole keyword should have a slash termination - i.e. something like:
```
PERMFACT
   0    0
   0.5 0.5
   1    1
   1.5  1.5  /  -- Terminate the first record
/                   -- Terminate the whole keyword.
```
I do not know how this should be used - if it is *always* only one record you could add `"size" : 1` to the `PERMFACT` json scheme?

File: tests/parser/SaltTableTests.cpp [hunk 2]
```diff
@@ -53,26 +56,40 @@ BOOST_AUTO_TEST_CASE( Brine ) {
         "1 1/\n"
         "\n"
         "PVTWSALT\n"
-        " 1000/\n"
```
Comment (joakim-hove) [inline tests/parser/SaltTableTests.cpp:56]: When you are making changes to the testdata you could consider using the support for multiline string literals in C++:
```
std::string deckData = R"(
Her comes the 
deck;

without any \n
or " characters

much easier to work with!
)";
```

**Comment by joakim-hove** [comment]
When the test failure has been fixed this can be merged.

**Comment by joakim-hove** [comment]
Ping?

File: tests/parser/SaltTableTests.cpp (hunk 1 above)
Comment (goncalvesmachadoc) [reply tests/parser/SaltTableTests.cpp:67]: Thanks for catching it. Multiples tables should be allowed, I have added the extra slash.

File: tests/parser/SaltTableTests.cpp (hunk 2 above)
Comment (goncalvesmachadoc) [reply tests/parser/SaltTableTests.cpp:56]: Thanks for the tip! :)

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by goncalvesmachadoc** [comment]
@joakim-hove Do you have any tips on how can I tackle this PR so it can pass the tests here?

**Comment by joakim-hove** [comment]
Ping me tomorrow and we can have a look together. I assume ut fails Aldo om your machine?

**Comment by bska** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
OK - this has been long in the making; it is green now - I'll merge. Good effort!

**Pull request merged**